### DISCUSSION TITLE
How to get block confirmations after submiting job to stratum API?

### DISCUSSION THREAD

**Discussion by blackmennewstyle** [BeamMW/beam#1931] [Q&A] [answered]
Hello,

I'm trying to work with https://github.com/BeamMW/beam/wiki/Beam-mining-protocol-API-(Stratum)

I have a question regarding `block confirmations` when i submit a `solution` to the `stratum API`.

If i submit a `solution` and the `stratum API` accepts it and then returns `accepted` and `blockHash`. Does it always mean i have won that block?

Also what about `block minimum confirmation`? Does `BEAM` has a `block confirmations` mechanism? If yes, is there any RPC method which will return that value?

**Comment by valdok** [comment, marked as answer]
Most probably it means you won the block, but not necessarily. There can be a race condition (someone could've found the solution just a little earlier). There can also be a reorg in the future.

The most robust way is to see of the block with this hash remain in the blockchain after a specific time interval.

**Reply by blackmennewstyle** [reply]
Thank you :)

I made some digging and indeed comparing the `blockhash` is the most accurate indicator. I also read about the necessary`240` block confirmations.

Have a great week ahead :fire: :rocket:
